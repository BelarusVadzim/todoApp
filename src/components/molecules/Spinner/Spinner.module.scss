@import 'src/variables.scss';

.overlay {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
}

.overlayHidden {
  @extend .overlay;

  display: none;
}

.spinner {
  position: absolute !important;
  top: 30% !important;
  left: 50% !important;
}

