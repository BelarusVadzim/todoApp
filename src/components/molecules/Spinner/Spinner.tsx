import { FadeLoader } from 'react-spinners';
import style from './Spinner.module.scss';

type SpinnerProps = {
  visible?: boolean;
};

const Spinner: React.FC<SpinnerProps> = ({ visible }) => {
  const overlayStyle = visible ? style.overlay : style.overlayHidden;

  return (
    <div className={overlayStyle}>
      <FadeLoader
        className={style.spinner}
        color={'#1064ae'}
        loading
        aria-label="Loading Spinner"
        data-testid="loader"
      />
    </div>
  );
};

export default Spinner;
