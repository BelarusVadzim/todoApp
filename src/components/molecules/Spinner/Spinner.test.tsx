import { render } from '@testing-library/react';
import Spinner from './Spinner';

afterEach(() => {
  jest.clearAllMocks();
});

describe('<Spinner />', () => {
  it('should render properly', () => {
    const component = render(<Spinner visible={true} />);

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('when visible is false should render properly ', () => {
    const component = render(<Spinner />);

    expect(component.asFragment()).toMatchSnapshot();
  });
});
