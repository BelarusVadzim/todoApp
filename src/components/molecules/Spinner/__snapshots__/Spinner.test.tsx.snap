// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> should render properly 1`] = `
<DocumentFragment>
  <div
    class="overlay"
  >
    <span
      aria-label="Loading Spinner"
      class="spinner"
      data-testid="loader"
      style="display: inherit; position: relative; font-size: 0px; top: 20px; left: 20px; width: 60px; height: 60px;"
    >
      <span
        style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(16, 100, 174); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.12s infinite ease-in-out; top: 20px; left: 0px;"
      />
      <span
        style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(16, 100, 174); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.24s infinite ease-in-out; top: 13.636363636363637px; left: 13.636363636363637px; transform: rotate(-45deg);"
      />
      <span
        style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(16, 100, 174); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.36s infinite ease-in-out; top: 0px; left: 20px; transform: rotate(90deg);"
      />
      <span
        style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(16, 100, 174); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.48s infinite ease-in-out; top: -13.636363636363637px; left: 13.636363636363637px; transform: rotate(45deg);"
      />
      <span
        style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(16, 100, 174); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.6s infinite ease-in-out; top: -20px; left: 0px;"
      />
      <span
        style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(16, 100, 174); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.72s infinite ease-in-out; top: -13.636363636363637px; left: -13.636363636363637px; transform: rotate(-45deg);"
      />
      <span
        style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(16, 100, 174); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.84s infinite ease-in-out; top: 0px; left: -20px; transform: rotate(90deg);"
      />
      <span
        style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(16, 100, 174); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.96s infinite ease-in-out; top: 13.636363636363637px; left: -13.636363636363637px; transform: rotate(45deg);"
      />
    </span>
  </div>
</DocumentFragment>
`;

exports[`<Spinner /> when visible is false should render properly  1`] = `
<DocumentFragment>
  <div
    class="overlayHidden"
  >
    <span
      aria-label="Loading Spinner"
      class="spinner"
      data-testid="loader"
      style="display: inherit; position: relative; font-size: 0px; top: 20px; left: 20px; width: 60px; height: 60px;"
    >
      <span
        style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(16, 100, 174); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.12s infinite ease-in-out; top: 20px; left: 0px;"
      />
      <span
        style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(16, 100, 174); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.24s infinite ease-in-out; top: 13.636363636363637px; left: 13.636363636363637px; transform: rotate(-45deg);"
      />
      <span
        style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(16, 100, 174); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.36s infinite ease-in-out; top: 0px; left: 20px; transform: rotate(90deg);"
      />
      <span
        style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(16, 100, 174); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.48s infinite ease-in-out; top: -13.636363636363637px; left: 13.636363636363637px; transform: rotate(45deg);"
      />
      <span
        style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(16, 100, 174); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.6s infinite ease-in-out; top: -20px; left: 0px;"
      />
      <span
        style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(16, 100, 174); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.72s infinite ease-in-out; top: -13.636363636363637px; left: -13.636363636363637px; transform: rotate(-45deg);"
      />
      <span
        style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(16, 100, 174); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.84s infinite ease-in-out; top: 0px; left: -20px; transform: rotate(90deg);"
      />
      <span
        style="position: absolute; width: 5px; height: 15px; margin: 2px; background-color: rgb(16, 100, 174); border-radius: 2px; transition: 2s; animation-fill-mode: both; animation: react-spinners-FadeLoader-fade 1.2s 0.96s infinite ease-in-out; top: 13.636363636363637px; left: -13.636363636363637px; transform: rotate(45deg);"
      />
    </span>
  </div>
</DocumentFragment>
`;
